src/test/parse-fail/issue-33569.rs

```diff
@@ -0,0 +1,15 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// compile-flags: -Z no-analysis
+
+macro_rules! foo {
+  { $+ } => () //~ ERROR: expected identifier, found `+`
+}              //~^ ERROR: no rules expected the token `}`
```

Comment:
> the second error message should not be emitted at all

Agreed. I believe we can avoid the second error message entirely by replacing [this line](https://github.com/rust-lang/rust/blob/b3583193531c124e36aa1813cf5f8b8e667d016c/src/libsyntax/parse/parser.rs#L2705) with
```rust
self.parse_ident().unwrap_or_else(|mut e| {
    e.emit();
    keywords::Invalid.ident()
})
```
instead of modifying the `unreachable!` match arm in `parse_token_tree`.